types/node/index.d.ts

```diff
@@ -2352,6 +2352,29 @@ declare module "net" {
     import * as stream from "stream";
     import * as events from "events";
 
+    export interface SocketConstructorOpts {
+        fd?: number;
+        allowHalfOpen?: boolean;
+        readable?: boolean;
+        writable?: boolean;
+    }
+
+    export interface TcpSocketConnectOpts {
+        port: number;
+        host?: string;
+        localAddress?: string;
+        localPort?: number;
+        hints?: number;
+        family?: number;
+        lookup?: Function;
+    }
+
+    export interface IpcSocketConnectOpts {
+        path?: string;
```

Comment:
According to doc the path is required for IPC connections. If I understood the code correct the presence of path property is used to decide if TCP or IPC should be used.